rct229/rulesets/ashrae9012019/section5/section5rule43.py

```diff
@@ -0,0 +1,118 @@
+            rmrs_used=UserBaselineProposedVals(True, False, True),
+            each_rule=Section5Rule43.BuildingRule(),
+            index_rmr="proposed",
+            id="5-43",
+            description="The proposed roof surfaces shall be modeled using the same solar reflectance as in the user "
+            "model if the aged test data are available, or equal to 0.7 default reflectance",
+            ruleset_section_title="Envelope",
+            standard_section="Section G3.1-1(a) Building Envelope Modeling Requirements for the Proposed design",
+            is_primary_rule=True,
+            list_path="ruleset_model_instances[0].buildings[*]",
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule43.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(True, False, True),
+                each_rule=Section5Rule43.BuildingRule.RoofRule(),
+                index_rmr="baseline",
+                list_path="$.building_segments[*].zones[*].surfaces[*]",
+            )
+
+        def create_data(self, context, data=None):
+            building_p = context.proposed
+            return {
+                "scc_dict_p": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building_p
+                ),
+            }
+
+        def list_filter(self, context_item, data=None):
+            surface_p = context_item.proposed
+            return get_opaque_surface_type(surface_p) == OST.ROOF
+
+        class RoofRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule43.BuildingRule.RoofRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(True, False, True),
+                    required_fields={
+                        "$": ["surface_optical_properties"],
+                        "surface_optical_properties": ["absorptance_solar_exterior"],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                roof_p = context.proposed
+                roof_u = context.user
+                scc_dict_p = data["scc_dict_p"]
+
+                return {
+                    "absorptance_solar_exterior_p": roof_p[
+                        "surface_optical_properties"
+                    ]["absorptance_solar_exterior"],
+                    "absorptance_solar_exterior_u": roof_u[
+                        "surface_optical_properties"
+                    ]["absorptance_solar_exterior"],
+                    "surface_conditioning_category_p": scc_dict_p[roof_p["id"]],
+                }
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                return (
+                    calc_vals["absorptance_solar_exterior_p"]
+                    == calc_vals["absorptance_solar_exterior_u"]
```

Comment:
@weilixu This looks like it is giving PASS outcome for when absorptance_solar_exterior_p == absorptance_solar_exterior_u but absorptance_solar_exterior_p != ABSORPTANCE_SOLAR_EXTERIOR. The desired outcome for this case is UNDETERMINED.

Based on Maria's comments on the RDS, we can simplify the PASS condition to be when absorptance_solar_exterior_p == ABSORPTANCE_SOLAR_EXTERIOR.

Only the PASS message will depend on if absorptance_solar_exterior_u is also equal.